src/components/Books.js

```diff
@@ -65,21 +77,46 @@ const removeBook = async (title) => {
 
 const addBooks = async ({title, authors, description, published, image, link}) => {
   try{
-    console.log("USER",user);
-      const book = {title, authors, description, published, image, link, userId:user.attributes.sub}
-      let response = await API.graphql(graphqlOperation(createBook, {input:book}))
-      fetchBooks()
-      toast.success(`${title} Added Successfully`);
-      console.log(response);
+    console.log("USER",userId);
+      const book = {title, authors, description, published, image, link}
+      let  allBooks = await API.graphql(graphqlOperation(listBooks,{
+          filter : {
+            title : {
+              eq: title
+            }
+          }        
+      }))
+      console.log(allBooks.data.listBooks.items[0]);
+      console.log(user.attributes.sub);
+
+      if(allBooks.data.listBooks.items.length > 0){
+        let response = await API.graphql(graphqlOperation(createUserBooks, {
+          input: {
+            userBooksUserId: userId,
+            userBooksBookId: allBooks.data.listBooks.items[0].id
+          }
+        }))
+        console.log(response);          
+      }
+      else{
+        
+          let response = await API.graphql(graphqlOperation(createBook, {input:book}))
+          console.log(response.data.createBook.id);
+          let response_ = await API.graphql(graphqlOperation(createUserBooks, {
```

Comment:
Use const here. Also for response identifiers, you can use operationResponse name format. 
For eq: For this: 
```
const createUserBooksResponse = await ....
```

Reply:
Okay will follow this naming Covention